### PR TITLE
Add `objective` Flag And use `SuplusFeesCosts` for Price Estimates

### PR DESCRIPTION
**Pull request by nlordell**
This PR adds a new field to `SolverConfig` to allow configuring Quasimodo for different environments. Specifically, requests to Quasimodo from the solver should use a different objective then when requesting price estimates (when solving, Quasimodo defaults to optimizing with capped surplus, which we don't want for price estimates).

### Test Plan

Run the orderbook locally and see the new parameter:
```
% cargo run -p orderbook -- --price-estimators Quasimodo --log-filter shared::http_solver=trace
...
2022-05-25T13:26:42.718Z TRACE shared::http_solver: request url=https://.../solve?instance_name=2022-05-25_13%3A26%3A42.718259_UTC_Ethereum___Mainnet_1_0&time_limit=2&max_nr_exec_orders=100&use_internal_buffers=false&objective=surplusfeescosts body=...
...
```

### CODE REVIEW

File: crates/orderbook/src/main.rs [hunk 1]
```diff
@@ -38,7 +38,7 @@ use shared::{
     },
     baseline_solver::BaseTokens,
     current_block::current_block_stream,
-    http_solver::{DefaultHttpSolverApi, SolverConfig},
+    http_solver::{DefaultHttpSolverApi, OptimizeFor, SolverConfig},
```
Comment (twalth3r) [inline crates/orderbook/src/main.rs]: I would probably call this `Objective` instead of `OptimizeFor` in order to make the connection to the respective solver parameter more obvious.

**Comment by nlordell** [comment]
Oops 🙈 - fixed.

**Review by twalth3r — COMMENTED** [review]
Shouldn't it better be `objective` in all places where you wrote `optimize` or `OptimizeFor`?

File: crates/shared/src/http_solver.rs [hunk 2]
```diff
@@ -105,11 +126,24 @@ impl HttpSolverApi for DefaultHttpSolverApi {
                 use_internal_buffers.to_string().as_str(),
             );
         }
+        match self.config.objective {
+            Some(Objective::CappedSurplusFeesCosts) => {
+                url.query_pairs_mut()
+                    .append_pair("optimize", "cappedsurplusfeescosts");
```
Comment (twalth3r) [inline crates/shared/src/http_solver.rs:132]: There's another `optimize` here!

File: crates/orderbook/src/main.rs [hunk 3]
```diff
@@ -38,7 +38,7 @@ use shared::{
     },
     baseline_solver::BaseTokens,
     current_block::current_block_stream,
-    http_solver::{DefaultHttpSolverApi, SolverConfig},
+    http_solver::{DefaultHttpSolverApi, Objective, SolverConfig},
```
Comment (harisang) [inline crates/orderbook/src/main.rs:41]: Is this change needed here, or is the new parameter part of `SolverConfig`?

File: crates/orderbook/src/main.rs (hunk 1 above)
Comment (nlordell) [reply crates/orderbook/src/main.rs]: 👍 - renamed

File: crates/shared/src/http_solver.rs (hunk 2 above)
Comment (nlordell) [reply crates/shared/src/http_solver.rs]: Yes... Sorry, I thought I grepped everywhere. I missed these two places.

File: crates/orderbook/src/main.rs (hunk 3 above)
Comment (nlordell) [reply crates/orderbook/src/main.rs:41]: Its a new parameter part of `SOlverConfig`.

Driver creates one with default objective (so nothing specified in the query URL) for solving batches and the orderbook creates on with objective set to `surplusfeescosts` for price estimates.

**Comment by nlordell** [comment]
Sorry @twalth3r, so many brain-farts. Hopefully should be correct now.

**Review by twalth3r — APPROVED** [review]
Looks good to me now - thanks @nlordell !

**Review by sunce86 — APPROVED** [review]
Nice. Lgtm
